### ISSUE TITLE
Screen reading shader + Light2D rendered incorrectly on GLES2

### ISSUE THREAD

**Issue by djrain** [godotengine/godot#42521]
**Godot version:**
3.2.2 stable

**OS/device including version:**
MacBook Pro (Retina, 15-inch, Mid 2015)
MacOS Mojave 10.14.5 
Intel Iris Pro 1536 MB
GLES2

**Issue description:**
I had a scene with some water (a ViewportContainer) with a shader that uses SCREEN_TEXTURE for a distortion effect. Also in this scene is a Light2D that affects the water. I noticed that the area where the light and water overlap was not being rendered properly; there is some sort of flipped second copy of the light's texture (or entire screen?), visible within the light:

![Screen Shot 2020-10-02 at 5 44 18 PM](https://user-images.githubusercontent.com/33777501/94979352-f7b0be80-04d6-11eb-9bfd-38c1aa02dc7f.png)

The bug is **not** present in GLES3.

**Steps to reproduce:**
1. Make a new scene containing a Light2D and a TextureRect as siblings, order does not matter
2. Assign the Light2D and TextureRect some texture
3. Give the TextureRect a CanvasItem shader that reads the screen:
```
void fragment() {
	COLOR = mix(vec4(0.0, 0.6, 1.0, 1), texture(SCREEN_TEXTURE, SCREEN_UV), 0.9);
}
```
4. Position the Light2D overlapping the TextureRect, and notice the incorrect rendering within the Light2D. The issue is most obvious when you zoom in on the light.

**Minimal reproduction project:**
[ScreenReading-Light2D-Bug.zip](https://github.com/godotengine/godot/files/5320842/ScreenReading-Light2D-Bug.zip)

**Comment by clayjohn** [comment]
I did some searching and this may be a limitation of GLES2. :/ Right now the light texture and the screen_texture use the same texture slot. So it works fine when only one or the other is used, but when both are used, the light texture overwrites the screen_texture during the light pass and messes things up. 

Introduced here: https://github.com/godotengine/godot/commit/a366d458565f36856ec29b1f555edb8ecd7e16df

In the meantime you can use ``if (!AT_LIGHT_PASS)`` to disable the texture read when at a light pass. It won't be quite accurate, but it should be good enough as a workaround.

**Comment by clayjohn** [comment]
Closed by #42538
